### ISSUE TITLE
mongoose 4.4.4 : findOneAndUpdate : $set not working when value is map {}

### ISSUE THREAD

**Issue by ashishsonone** [Automattic/mongoose#3894]
mongoose 4.4.4
node v5.4.0
mongo db version v3.2.1

When using findOneAndUpdate with $set, to set value of a field whose value is a map {}, it returns an non-null but empty error object (ie. { } )
It works in mongoose 4.3.7. The code to reproduce the error is following
``` js
var mongoose = require('mongoose');
mongoose.connect("mongodb://localhost:27017/bug");

var ContentSchema = mongoose.Schema({
    _id : false,
    heading : String,
    tag : String
});

var newsSchema = mongoose.Schema({
    status : String,
    content : ContentSchema
});

var newsModel = mongoose.model('news', newsSchema, 'news');

var id = null;
function create(){
    var news = new newsModel();
    news.status = "uploaded";
    news.content = {
        heading : "hi",
        tag : "sports"
    };
    return news.save();
}

function setStatus(){
    var promise = newsModel.findOneAndUpdate({_id : id},
            {'$set' : {status : "approved"}},
            {new : true}).exec();
    return promise;
}

function setContent(){
    var content = {
        heading : "hi there",
        tag : "nothing"
    };
    var promise = newsModel.findOneAndUpdate({_id : id},
            {'$set' : {content : content}},
            {new : true}).exec();
    return promise;
}

function start(){
    create()
    .then(function(result){
        id = result._id;
        console.log("created %j" + result);
        return setStatus();
    })
    .then(function(result){
        console.log("updated status %j", result);
        return setContent();
    })
    .then(function(result){
        console.log("updated heading %j", result);
    })
    .then(null, function(err){
        console.log("error %j", err);
    });
}

setTimeout(start, 1000);
```

**Comment by vkarpov15** [comment]
This was fixed in 4.4.5, likely along with #3889 but I haven't bisected it down to an individual commit. Upgrade and it should work :)

**Comment by ashishsonone** [comment]
thanks :)
